tests/components/ios/test_init.py

```diff
@@ -26,6 +26,8 @@ def mock_dependencies(hass):
 
 async def test_creating_entry_sets_up_sensor(hass):
     """Test setting up iOS loads the sensor component."""
+    # so it will not have an empty config
+    await bootstrap.async_from_config_dict({"foo": "bar"}, hass)
```

Comment:
Why is this necessary?

Reply:
otherwise it has an empty config and it asserts

Reply:
This test is weirdly written. It's basically just testing the core. It would make more sense to create a mock config entry and set that up while patching the `hass.config_entries.async_forward_entry_setup` function and check that that is called.

Reply:
i didn't write the test, just tried to make the test itself throw an exception. I do agree that the only thing it may check is that discovery_flow was registered or something like this but the real code for the ios platform is all mocked in it.
i did see that this test exists quite identically in other components (e.g. tplink/test_init.py), so i assumed that this is the reason it is there. The only thing different about ios is that it tries to load notify and that throws an assertion on empty config.
Perhaps a better way to handle it is to add to the conftest.py hass fixture something that has some dummy config in it so it isn't empty. Alternatively, the assertion of non-empty config may be redundant, as most components do not use that config and if they do, they can check themselves. 
I don't have an opinion about whether the test should be there or not, as i am not the maintainer of ios and cannot really say why it was there (also, i never used the discovery flow). However, it probably doesn't make sense that the behavior in the test is different from the real life scenario in a way that wasn't explicitly planned, so i added this.
perhaps we put this in, as it is better IMO to have the test working as expected than the current level of the code where the test is there but the component behaves differently. In parallel, perhaps reach out to the maintainer of ios and have the dialogue about whether the test itself should be changed?

Reply:
If we want to test that the sensor platform is set up if the ios component is set up, we can do what I wrote above. That's enough.

Reply:
just pushed something. Let me know if this is what you meant

Reply:
```suggestion
```

Reply:
made these changes, but apparently if i use async_setup_component, i don't need the config entry to get there.
in any case, it looks like for some reason, this is also ignored from coverage. may take a stab at fixing this later and increasing coverage as i plan to use ios...

Reply:
please let me know if what i have there works in the latest push